lib/game_console.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require_relative '../dependencies'
+
+class GameConsole
+  include Validation
+  include Database
+  include Output
+
+  HINT = 'hint'
+  SAVE = 'save'
+  INPUT_DATA = /^[1-6]{4}$/.freeze
+
+  def initialize(name, difficulty)
+    @game = Game.new(name: name, difficulty: difficulty)
+  end
+
+  def start
+    loop do
+      break if @game.attempts.zero? || @game.win
+
+      start_info(@game.attempts, @game.hints)
+      input = gets.chomp
+      case input
+      when GameMenu::EXIT then break close
+      when HINT then next puts @game.use_hint
+      when INPUT_DATA then puts @game.check(input)
+      else puts I18n.t(:wrong_process)
+      end
+    end
+    puts I18n.t(:game_over)
+    statistics
+  end
+
+  def statistics
```

Comment:
greedy method: it shows statistics and saves data
move:
```
if @game.win
      puts I18n.t(:win)
      puts I18n.t(:save)
      save_results if gets.chomp == SAVE
    else
      puts I18n.t(:lose)
    end
```
to another method, maybe `end_game`

Reply:
created end_game method and moved what you asked.